scripts/tests/twister/test_config_parser.py

```diff
@@ -0,0 +1,265 @@
+    assert scenario_data["toolchain_allow"] == set()
+    assert scenario_data["filter"] == ""
+    assert scenario_data["levels"] == []
+    assert scenario_data["harness"] == "test"
+    assert scenario_data["harness_config"] == {}
+    assert scenario_data["seed"] == 0
+    assert scenario_data["sysbuild"] is False
+
+
+def test_cast_string_to_string(zephyr_base):
+    value = "hello"
+    typestr = "str"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == "hello"
+
+
+def test_cast_string_to_float(zephyr_base):
+    value = "3.14"
+    typestr = "float"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == 3.14
+
+
+def test_cast_string_to_int(zephyr_base):
+    value = "10"
+    typestr = "int"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == 10
+
+
+def test_cast_string_to_bool(zephyr_base):
+    value = True
+    typestr = "bool"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result is True
+
+
+def test_cast_string_to_map(zephyr_base):
+    value = "map1 map2"
+    typestr = "map"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == "map1 map2"
+
+
+def test_raises_error_invalid_cast(zephyr_base):
+    value = "test"
+    typestr = "int"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+
+    with pytest.raises(ValueError):
+        parser._cast_value(value, typestr)
+
+
+def test_cast_string_to_unknown_type(zephyr_base):
+    value = "test"
+    typestr = "unknown"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+
+    with pytest.raises(ConfigurationError):
+        parser._cast_value(value, typestr)
```

Comment:
You can achieve analogous testing in more concise and simpler to expand way, using parametrisation:
```python
@pytest.mark.parametrize(
    'value, typestr, expected',
    [
        (' hello ', 'str', 'hello'),
        ('3.14', 'float', 3.14),
        ('10', 'int', 10),
        ('True', 'bool', 'True'),          # do-nothing cast
        ('key: val', 'map', 'key: val'),   # do-nothing cast
        ('test', 'int', ValueError),
        ('test', 'unknown', ConfigurationError),
    ],
    ids=['str to str', 'str to float', 'str to int', 'str to bool', 'str to map',
         'invalid', 'to unknown']
)
def test_cast_value(zephyr_base, value, typestr, expected):
    loaded_schema = scl.yaml_load(
        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
    )

    parser = TwisterConfigParser("config.yaml", loaded_schema)

    with pytest.raises(expected) if \
         isinstance(expected, type) and issubclass(expected, Exception) else nullcontext():
        result = parser._cast_value(value, typestr)
        assert result == expected
```

Reply:
Additionally, where is `typestr = list` and `typestr = set` testing?

Reply:
neet :)
list and set defeated me

Reply:
+list and set casting test